### PR TITLE
New reaction: Nitrous Oxide decomposition

### PR DESCRIPTION
**Pull request by 0xJoey**
## About The Pull Request

N2O is a fairly good oxidizer in real life, why not in this game? This PR adds a new reaction that causes N2O to exothermically decompose into N2 and O2, ~~starting slowly from 1400K but rapidly speeding up as it approaches ~6400K.~~ starting from 1400, reaching a peak at 50000K and stopping completely at 100000K. This is in order to not create any problems with nitryl formation and fusion.

~~Because this causes problems with the SM, I've changed the SM to use Nitryl as a safety gas, as it has no other uses other than being an intermediate product and as a shitty zoom zoom gas. A canister is placed in the engine room at roundstart. You don't technically need Nitryl, but it's good for when things go awry.~~

Don't put it in a hot engine, ~~unless you like fun~~.

## Why It's Good For The Game

More gas reactions causes more complex interactions. Furthermore, this makes N2O reflect its real-world use in a better way.

This should not make it easier to get hot fires. Pure O2 with Plasma is still the best way to get hot fires.

## Changelog
:cl: JoeyJo0
add: N2O Decomposition reaction
tweak: N2O is no longer the perfect "safety gas". N2O Decomposes starting from 1400K, feeding more oxygen into the engine. 
/:cl:

### CODE REVIEW

**Comment by RaveRadbury** [comment]
Are the days of unwrenching the cooling loop and releasing laughing gas truly over?

**Comment by zxaber** [comment]
You should leave N2O for the engine. Having a gas that works great for cooling under normal conditions but makes things worse during emergencies is neat. Anyway, we still have N2 for times when the SM is on fire. (And really, the reaction will take care of that for us, eventually).

**Comment by OverlyGeneric** [comment]
> You should leave N2O for the engine. Having a gas that works great for cooling under normal conditions but makes things worse during emergencies is neat. Anyway, we still have N2 for times when the SM is on fire. (And really, the reaction will take care of that for us, eventually).

The only thing I'm concerned about replacing N2O is replacing it with NO2. 

Let's not give the powergamers another item to check off their list.

**Comment by Whoneedspacee** [comment]
Raw nitryl is pretty garbage for many reasons.

1. Stimulum production is really what you want nitryl for. As far as I know fusion is the only effective way to get any real amount of stimulum (though I’m inexperienced with new fusion)

2. For stimulum you still need multiple other gasses that are a major pain in the ass to get, as well as the heat necessary, and you produce barely anything. If anything this might make it slightly viable as before the best stimulum speedrun without fusion was around the 50 minute mark.

3. Nitryl damages you, why even use it when...

4. Meth, makes you move faster and doesn’t terribly fuck up your lungs, and much easier to get. Don’t even get me started on optimizing air tanks to work around shitty breath code to get the benefits of nitryl / stim without suffocating yourself.

tl;dr probably won’t be useful for powergaming and adds a new layer of complexity to atmos, I’m all for it.

**Comment by 0xJoey** [comment]
Well, that's what I was worried about, so you confirmed my suspicions. Not sure how much this is going to fuck over the engine once it's really hot, though.

**Comment by Barhandar** [comment]
@zxaber the whole _point_ of N2O is that it's emergency gas; it's not useful during regular running at all (unless you have extremely specific setups, in which case they qualify as chronic emergency anyway) because it ruins energy output. N2 already does the regular coolant role.

**Review by duncathan — COMMENTED** [review]
could you explain briefly how this differs from #9987? that particular implementation was troublesome for a couple of reasons but I would like to hear what you think are the major differences with yours.

cc @as334

**Comment by anconfuzedrock** [comment]
also don't like the free can being placed in the engine room. maybe it'd be interesting if people had to work to get the "literally stops engineering from exploding and giving everyone nerfed hallucinations" gas, but no, let's not force atmos techs to actually mix any useful gas.

**Comment by Whoneedspacee** [comment]
wait how does n2o mess up energy output

last i checked n2 was the gas that fucked up energy output

**Comment by 0xJoey** [comment]
@duncathan Real-world oxidation by N2O is a result of the decomposition of N2O into N2 and O2. This is to better reflect that. N2O is often used as an oxidizer because it's easier to handle and store, not because of its great qualities as an oxidizer.

This just breaks down N2O above a certain temperature, giving the fire more O2, but also adding N2. I'm pretty sure tweaks will still be required.

**Comment by as334** [comment]
This will make fusion either not work or at least severely limit its effectiveness. The web of reactions interfering with each other is complicated, but this would cause the waste N2O from fusion to break down and produce N2, which would then interact with the tritium to both suck out energy and produce Noblium which would eventually choke out the reaction.

**Comment by zxaber** [comment]
@Barhandar We glanced at the code, and as far as we can tell N2 lowers heat directly just by existing (which also lowers energy output), while N2O protects against higher heats. That should mean that N2O is great for running riskier engine setups, or for squeezing out more power from a standard setup, and N2 is a safe gas that helps slow down engine overheating in the first place.

As such, this change would make N2O good for pushing out more power, but riskier because if things start going south it will just add to the problem. If you ask me, it's a pretty neat trade-off and brings a bit more depth to the choice of which coolant to run. Default engine setups are set to use N2 anyway, so this should only affect engineers that customize their engine.

**Comment by Fox-McCloud** [comment]
Not a big fan of making it decompose--what if I want to knock out people with blazing hot N2O?

I realize it's still technically possible, what with the cutoffs, but ehhhhh.

**Comment by Unit2E** [comment]
To clear up some SM related troubles people seem to have, N2O is the emergency gas, N2 is the normal coolant gas. Why is this the case? N2 reduces 'heat penalty', which reduces waste and heat build up from running the SM as whole. The reason a pure plasma setup lights the engine on fire is because it has a huge heat penalty, which is getting the SM to produce a ton of oxygen and plasma, and a ton of heat. Reducing heat penalty, as such, will make the SM slower to catch fire, which is why N2 is used.

Why is N2O the emergency gas? It reduces the damage from heat, by a lot. The difference is that, if the engine is 2000K hot, N2 lowering how fast it gets hot hardly matters, but reducing the direct damage something does is a big deal. Using N2O means an engine can run to be about stable at much higher temperatures, which isn't relevant for 99% of engine setups, aside from plasma engines. However, it is extremely handy for when a normal engine is delamming, as it would drastically decrease the speed of delamination, even so far as reducing it to 0%. On that note, the reason the delamination rate is 0% sometimes is often up to N2O, the SM is, generally, in a state of either taking damage or being stable / regaining structure. When it is taking damage, or would normally take damage, it will not become stable. N2O reduces damage, but it does not shift the treshold for becoming stable, so you end up in the damaging part of the SM, but without the SM actually taking any damage. This can also happen with N2, as there is some gap between recovery and damaging, as well, but N2O really makes it very noticeable.

As for power generation, N2O does not have a negative effect on power generation. N2, however, does. The effect is extremely minor, however, in fact, N2 may have positive effects on power generation overall. `gasmix_power_ratio = min(max(plasmacomp + o2comp + co2comp - n2comp, 0), 1)` tells us that the power ratio of the mix does decrease from N2, and, `power = max( (removed.temperature * temp_factor / T0C) * gasmix_power_ratio + power, 0)` tells us that temperature will generally have positive effects on power. In reality, the power difference between N2O and N2 are extremely minimal, not-even-noticeable at all levels.

More on topic now, as334 is somewhat likely to be right in this case, this will have a negative effect on fusion. Mind, their described scenario may not be extremely likely, it would take a very fair amount of N2 and tritium for this to happen. Especially since, with fusion, the tritium tends to react before it would even reach high enough, moles wise, to undergo the noblium reaction. And even if it does, nothing is really stopping you from filtering out noblium somehow. That being said, it is definitely something that may affect certain fusion setups, at the least when the person attempting fusion does not account for noblium creation, minor though it may be. I don't think this is reason enough not to merge the PR, though. As I mentioned, you can just filter it to have it not happen, which may just be another factor to account for with fusion, which I personally don't dislike.

That's all for me, peace

**Comment by 0xJoey** [comment]
@Unit2E Thanks for the clarification.

I've discussed the matter with @as334 and have decided to have a temperature window in which N2O decomposes. It's not the best solution, especially not because it was not what I originally planned, but it's a good solution until something better can be found(or, knowing how SS13 codebases work, be a permanent solution until an inevitable rewrite of the whole system).

**Comment by duncathan** [comment]
is there a replacement emergency coolant?

also, you will need a better changelog to make it clear what some of the ramifications of the change are, particularly re: the SM

**Comment by optimumtact** [comment]
time for FREON again

**Comment by Shdorsh** [comment]
mmh, not happy with this one, as one of my main setups for emitter-less SM is n2o and co2, if i remember correctly. still, most people just run the emitter normally.

**Comment by optimumtact** [comment]
@as334 @Unit2E opinions on this final state?

**Comment by as334** [comment]
If it's been changed to not interfere with fusion and such, it seems fine to me.

**Comment by Daxxed** [comment]
Nitryl req was lowered to 22000K from 2e^6K.
This makes it alot harder to farm nitryl as all n2o will dissapear within seconds unless you heat it to 100000K+.
